internal/buildpackage/builder_test.go

```diff
@@ -1,8 +1,8 @@
 package buildpackage_test
```

Comment:
This PR is built on top of #393 and hence once merged will be incorporated there. The reason they exist here is that it was through this work that we discovered that the tests were misplaced. If there are no major drawbacks I'd like to keep it here in order to prevent having to make additional changes to the original PR.